inf-ruby.el

```diff
@@ -314,26 +330,31 @@ in `inf-ruby-implementations'.
                                         inf-ruby-implementations))))
   (setq name (or name "ruby"))
 
-  (if (not (comint-check-proc inf-ruby-buffer))
+  (if (not (comint-check-proc (inf-ruby-buffer)))
       (let ((commandlist (split-string-and-unquote command))
             (buffer (current-buffer))
             (process-environment process-environment))
         ;; http://debbugs.gnu.org/15775
         (setenv "PAGER" (executable-find "cat"))
-        (set-buffer (apply 'make-comint name (car commandlist)
+        (set-buffer (apply 'make-comint-in-buffer
+                           name
+                           (generate-new-buffer-name (format "*%s*" name))
+                           (car commandlist)
                            nil (cdr commandlist)))
         (inf-ruby-mode)
-        (ruby-remember-ruby-buffer buffer)))
-  (pop-to-buffer (setq inf-ruby-buffer (format "*%s*" name))))
+        (ruby-remember-ruby-buffer buffer)
+        (add-to-list 'inf-ruby-buffers (current-buffer))))
```

Comment:
I was looking at this several times, but forgot to mention: please use `push` instead of `add-to-list`. It reads better, and will continue to work under lexical binding (which we'll surely switch to someday).